src/aggregator/aggregator/leader_flush_mgr.go

```diff
@@ -181,20 +190,69 @@ func (mgr *leaderFlushManager) Prepare(buckets []*flushBucket) (flushTask, time.
 		}
 	}
 
-	durationSinceLastPersist := time.Duration(nowNanos - mgr.lastPersistAtNanos)
-	if mgr.flushedSincePersist && durationSinceLastPersist >= mgr.flushTimesPersistEvery {
-		mgr.lastPersistAtNanos = nowNanos
-		mgr.flushedSincePersist = false
-		flushTimes := mgr.prepareFlushTimesWithLock(buckets, allShards)
-		mgr.flushTimesManager.StoreAsync(flushTimes)
-	}
-
 	if !shouldFlush {
+		// NB: if enough time has passed since last flush time persist, run the
+		// persist by itself.
+		if shouldUpdateFlushTimes {
+			mgr.flushTask.runNextOnCompleteTask = false
+			return mgr.flushTask.onCompleteTask, waitFor
```

Comment:
i'm having a really hard time following this case, where we have 2 different types of flush tasks and this extra boolean.

why don't we just sync this flush times here directly if we aren't going to run a flush task?

Reply:
We want to run the sync after the shard flush time sync if we've started a flush, but yeah this case can probably just perform the flush directly